pm/sender.go

```diff
@@ -32,15 +33,22 @@ type session struct {
 type sender struct {
 	signer        Signer
 	roundsManager RoundsManager
+	senderManager SenderManager
+
+	maxEV             *big.Rat
+	depositMultiplier int
 
 	sessions sync.Map
 }
 
 // NewSender creates a new Sender instance.
-func NewSender(signer Signer, roundsManager RoundsManager) Sender {
+func NewSender(signer Signer, roundsManager RoundsManager, senderManager SenderManager, maxEV *big.Rat, depositMultiplier int) Sender {
```

Comment:
Again a slightly out of place comment since this functionality was already in place before the PR (consider it a question), but shouldn't we be returning a concrete struct or pointer here, not an interface value, according to the [go wiki ] ?(https://github.com/golang/go/wiki/CodeReviewComments#interfaces)

> Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values. The implementing package should return concrete (usually pointer or struct) types

Reply:
Yeah returning a concrete type here instead of an interface defined by the `pm` package probably would be better. Another annoyance that arises from the current approach is that in unit tests we have to do a type assertion on the interface if we want to run checks on unexported fields.